index.php

```diff
@@ -94,10 +94,34 @@
         ));
     }
 
+    global $CFG;
+    if (!empty($CFG->rememberusername)) {
+        $cookiename = 'REMEMBERUNAME';
+        if (isset($_COOKIE[$cookiename])) {
+            $username = rc4decrypt($_COOKIE[$cookiename]);
+        }
+        global $frm;
+        $frm->username = $username;
+        $passwordform->set_data(array(
+            'username' => $username,
+            'rememberme' => 1,
+        ));
+    }
+
     if ($passwordform->is_submitted() &&
         $passwordform->is_validated()) {
         $values = $passwordform->get_data();
         $helper->set_passform_params_in_auth_global_vars($values);
+
+        //Setting loginas cookie
+        if (!empty($values->rememberme)) {
+            global $CFG;
+            // Set username cookie for 60 days.
+            $cookiesecure = is_moodle_cookie_secure();
+            $Loggedinusername = $values->username;
```

Comment:
can we directly pass is_moodle_cookie_secure(), $values->username values for this in setcookie function.
we don't need local variable for it.

Reply:
Done, Please review.